pkg/envinfo/envinfo_test.go

```diff
@@ -0,0 +1,247 @@
+package envinfo
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"testing"
+
+	"github.com/openshift/odo/pkg/testingutil/filesystem"
+
+	"github.com/openshift/odo/pkg/util"
+)
+
+func TestSetEnvInfo(t *testing.T) {
+
+	tempEnvFile, err := ioutil.TempFile("", "odoenvinfo")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tempEnvFile.Close()
+	os.Setenv(envInfoEnvName, tempEnvFile.Name())
+	testURL := ConfigURL{Name: "testURL", ClusterHost: "1.2.3.4.nip.io", TLSSecret: "testTLSSecret"}
+
+	tests := []struct {
+		name            string
+		parameter       string
+		value           interface{}
+		existingEnvInfo EnvInfo
+	}{
+		{
+			name:      fmt.Sprintf("Case 1: %s to test", URL),
+			parameter: URL,
+			value:     testURL,
+			existingEnvInfo: EnvInfo{
+				componentSettings: ComponentSettings{},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			esi, err := NewEnvSpecificInfo("")
+			if err != nil {
+				t.Error(err)
+			}
+			esi.EnvInfo = tt.existingEnvInfo
+			err = esi.SetConfiguration(tt.parameter, tt.value)
+			if err != nil {
+				t.Error(err)
+			}
+
+			isSet := esi.IsSet(tt.parameter)
+
+			if !isSet {
+				t.Errorf("the '%v' is not set", tt.parameter)
+			}
+
+		})
+	}
+}
+
+func TestLocalUnsetConfiguration(t *testing.T) {
```

Comment:
since the function above is TestSetEnvInfo, please change this to TestUnsetEnvInfo

Reply:
updated